### DISCUSSION TITLE
Administration - import and export configuration

### DISCUSSION THREAD

**Discussion by Gooseman** [ThreeMammals/Ocelot#1756] [Q&A]
There is a chapter in the documentation administration.  But the information given isn't sufficient to enable administration, specifically the import and export of configuration endpoints that are available, but that require some form of authentication.  The chapter says there are two options - an external identity server or the identity server bundled with Ocelot.  As a basic test, I'm trying to use the internal identity server.  However, all I get is a 400 (bad request) response.

I am trying with the Postman collection available in the Git repo.  I have changed the client_id to "admin", the client_secret to "secret" and scope to "admin" as instructe in the doc.

I have the web API app that is running Ocelot listening on https://localhost:5003 and http://localhost:5103.  The result is the same from both addresses.   It makes no difference whether the BaseUrl is set in the ocelot.json.GlobalConfiguration.

The application is a .net 7 web API running from the IDE (Rider), with Ocelot 20.0.0 and Ocelot.Administration 20.0.0.

I don't know what to look for to get this working.  This is just a test for now, but we will need to be able to use the configuration import and export in production, so this isn't just an exercise where the outcome doesn't matter.

**Comment by raman-m** [comment]
Hi Craigus!
Thanks for your interest in Ocelot!

Did you check our [AdministrationApi sample](../tree/main/samples/AdministrationApi) ?
It is a bit outdated because the reference to [v18](../blob/main/samples/AdministrationApi/AdministrationApi.csproj#L12) is used.

**Comment by Gooseman** [comment]
I did try it, but it didn't work.  I can try it again to see if I get a different outcome.

**Reply by raman-m** [reply]
Doesn't it work completely or you experienced some errors?
Were you be able to start [AdministrationApi sample](../tree/main/samples/AdministrationApi)?

**Reply by raman-m** [reply]
And also, it is time to migrate to Ocelot [v21](../releases/tag/21.0.0) because it targets .NET 6, 7, 8.
